pkg/build/util/generate.go

```diff
@@ -13,6 +14,7 @@ import (
 // GenerateBuildFromConfig creates a new build based on a given BuildConfig. Optionally a SourceRevision for the new
 // build can be specified.  Also optionally a list of image names to be substituted can be supplied.  Values in the BuildConfig
 // that have a substitution provided will be replaced in the resulting Build
+// You NEED to update BuildConfig object after calling this!
```

Comment:
@bparees Having thought about it a little bit more, the idea of accepting `BuildCOnfigUpdater` is bad in that if somebody update BC in the meantime I'm pushing his changes (intentionally or not). I'm proposing for this method to return new build object and it's number which then should be saved by the caller. 

@smarterclayton I'm puzzled with your suggestion, should be try to solve it or drop the idea of build naming with the incremental number at all?